### PR TITLE
written answers

### PR DESCRIPTION
**Pull request by SowmyaTani2000**

### CODE REVIEW

File: answers.txt [hunk 1]
```diff
@@ -1,4 +1,14 @@
 •  What is a gitignore file and what should it include for a Java project? 
+It is a text file which is placed in the git repository and conveys the git to not track certain files and folders that one doesn't want to upload to the master repository. .class, .template files that are created by ide at runtime.
+
```
Comment (josephtv) [inline answers.txt:4]: Could you please add little bit more about gitignore file? Also you need to specify what should it include for a Java project.

File: answers.txt [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+
 •  What does fast-forward in git mean? 
+It means to move the main branch’s tip to the end of one’s feature branch.
+
```
Comment (josephtv) [inline answers.txt:9]: Could you elaborate more on this, if possible with an example?

File: answers.txt [hunk 3]
```diff
@@ -1,4 +1,14 @@
...
+
 •  Why should you have different branches? 
-•  In a project how many remote Git repositories do you usually have?
\ No newline at end of file
+Branches allow the developer to branch out from the main line and work independently from others without messing with the main line.
```
Comment (josephtv) [inline answers.txt:12]: Could you explain how branches help keeping the main branch clean and always working?

**Review by josephtv — COMMENTED** [review]
Added comments

File: answers.txt (hunk 3 above)
Comment (SowmyaTani2000) [reply answers.txt:12]: The master branch should always be kept clean. When multiple developers are working, they simply create a branch independently and if they want to merge it with the master, they need to pull request. This way the master branch is not tampered in any way. The developers work on their feature branches.

File: answers.txt (hunk 2 above)
Comment (SowmyaTani2000) [reply answers.txt:9]: Updated the file

File: answers.txt (hunk 1 above)
Comment (SowmyaTani2000) [reply answers.txt:4]: Addressed.

**Review by josephtv — COMMENTED** [review]
Looks good. Approved.

**Review by josephtv — APPROVED** [review]
Looks good. Approved.

**Pull request merged**